stdlib/list.ml

```diff
@@ -56,6 +56,21 @@ let rec rev_append l1 l2 =
 
 let rev l = rev_append l []
 
+let rec init_tailrec_aux acc i n f =
+  if i >= n then acc
+  else init_tailrec_aux (f i :: acc) (i+1) n f
+
+let rec init_aux i n f =
+  if i >= n then []
+  else
+    let r = f i in
+    r :: init_aux (i+1) n f
+
+let init len f =
+  if len < 0 then invalid_arg "List.init" else
+  if len > 10_000 then rev (init_tailrec_aux [] 0 len f)
```

Comment:
This will stack-overflow when compiled to JavaScript with js_of_ocaml (because the javascript stack is much smaller). Any chance to lower the limit to ~100 ?

Reply:
Maybe the constant could depend on `Sys.backend_type` (cf https://github.com/ocaml/ocaml/blob/trunk/stdlib/sys.mli#L99 ) ?

Reply:
It's not wholly clear from its status, but this PR was merged, so please could this go in a new PR or in a Mantis report?

Reply:
https://github.com/ocaml/ocaml/pull/1697